### PR TITLE
Transmit the session ID as cookie instead of part of URL

### PR DESCRIPTION
**Pull request by Wadeck**
The code is only used for the testing.

Similar approach as the one used in BO: https://github.com/jenkinsci/blueocean-plugin/commit/fb75e524bc70b861d1819544e476b14f27ab43fa#diff-7c4e088720f8fe8962618f0947472140L101-R121

### CODE REVIEW

File: src/main/js/SSEConnection.js [hunk 1]
```diff
@@ -151,15 +151,16 @@ SSEConnection.prototype = {
                 var listenUrl = sseConnection.jenkinsUrl + '/sse-gateway/listen/'
                     + encodeURIComponent(tabClientId);
 
+                var options = undefined;
                 if (sseConnection.configuration.sendSessionId) {
```
Comment (dwnusbaum) [inline src/main/js/SSEConnection.js]: Ok, so in non-test code, this setting is false, so this is why this only matters for tests?

File: src/main/js/SSEConnection.js [hunk 2]
```diff
@@ -151,15 +151,16 @@ SSEConnection.prototype = {
...
                     // Sending the jsessionid helps headless clients to maintain
                     // the session with the backend.
                     var jsessionid = response.data.jsessionid;
-                    listenUrl += ';jsessionid=' + jsessionid;
+                    options = {headers: {Cookie: 'jsessionid=' + jsessionid}}
                 }
 
                 var EventSource = window.EventSource;
-                var source = new EventSource(listenUrl);
+                var source = new EventSource(listenUrl, options);
```
Comment (dwnusbaum) [inline src/main/js/SSEConnection.js]: Any problems with passing `undefined` here in non-test code?

Comment (Wadeck) [reply src/main/js/SSEConnection.js]: When you have a JavaScript method expecting 2 parameters where the second is optional, calling a(b) or a(b, undefined) is "similar". I tested manually this one to be sure ;)

File: src/main/js/SSEConnection.js (hunk 1 above)
Comment (Wadeck) [reply src/main/js/SSEConnection.js]: Essentially yes. It's required only for code that cannot handle the regular cookies (= test libraries in this case)

**Comment by olamy** [comment]
fyi all green in  BO https://ci.blueocean.io/blue/organizations/jenkins/blueocean/detail/sse-gateway-1.23/2/pipeline

**Pull request merged**